src/settings/plugins/keyfile/nms-keyfile-plugin.c

```diff
@@ -412,39 +429,49 @@ _sort_paths (const char **f1, const char **f2, GHashTable *paths)
 }
 
 static void
-read_connections (NMSettingsPlugin *config)
+_read_dir (GPtrArray *filenames, const char *path)
 {
-	NMSKeyfilePlugin *self = NMS_KEYFILE_PLUGIN (config);
-	NMSKeyfilePluginPrivate *priv = NMS_KEYFILE_PLUGIN_GET_PRIVATE (self);
 	GDir *dir;
-	GError *error = NULL;
 	const char *item;
-	GHashTable *alive_connections;
-	GHashTableIter iter;
-	NMSKeyfileConnection *connection;
-	GPtrArray *dead_connections = NULL;
-	guint i;
-	GPtrArray *filenames;
-	GHashTable *paths;
+	GError *error = NULL;
 
-	dir = g_dir_open (nms_keyfile_utils_get_path (), 0, &error);
+	dir = g_dir_open (path, 0, &error);
 	if (!dir) {
 		_LOGW ("cannot read directory '%s': %s",
-		             nms_keyfile_utils_get_path (),
+		             path,
```

Comment:
indentation

Reply:
Is this warning at all helpful? Now that we have multiple directories, it's not at all clear, that they all *must* exist.